### PR TITLE
Load chargebee subscriptions

### PR DESCRIPTION
**Pull request by fjvallarino**
@Rise-Vision/apps this is ready for review on stage-1. The only new commits are the last five, starting on 09ce518 (all the previous commits had already been merged to master when the parent branch was approved)

You can test with:

- Oleg's company: only the last Subscription on the list exists on Chargebee (the others will fail)
  - https://apps-stage-1.risevision.com/billing?cid=94b47745-faff-4c7e-8c82-51a56c732aba&cHJpb3JpdHktc3VwcG9ydA=1
- New company
  - https://apps-stage-1.risevision.com/billing?cid=a847f5bb-e61a-4925-817a-552075edf203&cHJpb3JpdHktc3VwcG9ydA=1

### CODE REVIEW

**Comment by fjvallarino** [comment]
@olegrise makes sense to me

**Comment by ezequielc** [comment]
@olegrise @fjvallarino Good idea. Thanks

File: web/scripts/billing/controllers/ctr-billing.js [hunk 1]
```diff
@@ -35,8 +63,35 @@ angular.module('risevision.apps.billing.controllers')
...
+      $scope.isSuspended = function (subscription) {
+        return subscription.status === 'Suspended';
+      };
+
+      function _getCurrency (subscription) {
+        return subscription.currencyCode.toUpperCase();
+      }
+
+      function _getPeriod (subscription) {
+        return subscription.unit === 'per Display per Month' ? 'Monthly' : 'Yearly';
```
Comment (olegrise) [inline web/scripts/billing/controllers/ctr-billing.js]: @fjvallarino I've noticed that all my monthly subscriptions are displayed as "Yearly". 
screen shot 2018-08-14 at 4 12 39 pm
My recommendation is to check that the `subscription.unit` contains `"per month"` ignoring letter case.

**Comment by olegrise** [comment]
@fjvallarino it doesn't look like the billing page handles company change events correctly. Is this something you going to address as part of this PR or it's going to be added later?

**Comment by fjvallarino** [comment]
@olegrise good point. I guess it's better to address it in this PR. I'll add it now

**Comment by fjvallarino** [comment]
@olegrise updated and deployed to stage-1

**Comment by alex-deaconu** [comment]
@fjvallarino we have a global handler for reloading lists on Company Change here: https://github.com/Rise-Vision/rise-vision-apps/blob/master/web/scripts/app.js#L458. Wondering if you could add your endpoint to that as well instead of a custom event. Thanks.

File: web/scripts/billing/controllers/ctr-billing.js [hunk 2]
```diff
@@ -2,10 +2,35 @@
...
-    'getCoreCountries', 'userState', 'chargebeeFactory', 'STORE_URL', 'INVOICES_PATH',
-    function ($scope, $loading, $window, $modal, $templateCache,
-      getCoreCountries, userState, chargebeeFactory, STORE_URL, INVOICES_PATH) {
+  .controller('BillingCtrl', ['$rootScope', '$scope', '$loading', '$window', '$modal', '$templateCache', '$timeout',
+    'ScrollingListService', 'getCoreCountries', 'userState', 'chargebeeFactory', 'billing', 'STORE_URL', 'INVOICES_PATH',
+    function ($rootScope, $scope, $loading, $window, $modal, $templateCache, $timeout, ScrollingListService,
+      getCoreCountries, userState, chargebeeFactory, billing, STORE_URL, INVOICES_PATH) {
+
+      $scope.search = {
+        sortBy: 'productName',
```
Comment (alex-deaconu) [inline web/scripts/billing/controllers/ctr-billing.js]: @fjvallarino what's the point of setting the sortBy parameter here if you have a static variable in the service? I would suggest moving the static value here and setting a boolean reverse. Also suggest the service code to be similar to other services:
`'sort': search.sortBy + (search.reverse ? ' desc' : ' asc')`

Thanks.

File: web/scripts/billing/services/svc-billing.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+              return storeApi.subscription.listUser(params);
+            })
+            .then(function (resp) {
+              $log.debug('susbcription.listUser resp', resp);
+
+              deferred.resolve(resp.result);
+            })
+            .then(null, function (e) {
+              console.error('Failed to get company\'s subscriptions.', e);
+              deferred.reject(e);
```
Comment (alex-deaconu) [inline web/scripts/billing/services/svc-billing.js]: @fjvallarino should we not handle and show this error to the User? Similar to: https://github.com/Rise-Vision/rise-vision-apps/blob/master/web/partials/displays/displays-list.html#L22. Thanks.

**Comment by fjvallarino** [comment]
@alex-deaconu updated!

**Comment by alex-deaconu** [comment]
@fjvallarino looks good, thanks for the fixes 👍

**Pull request merged**